### PR TITLE
vorta: enable tests and install a desktop file

### PR DESCRIPTION
**Pull request by Artturin**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/backup/vorta/default.nix [hunk 1]
```diff
@@ -16,19 +17,31 @@ python3.pkgs.buildPythonApplication rec {
...
+    qdarkstyle
+    secretstorage
+    appdirs
+    setuptools
   ];
 
-  # QT setup in tests broken.
-  doCheck = false;
+  postPatch = ''
+    sed -i -e '/setuptools_git/d' -e '/pytest-runner/d' setup.cfg
```
Comment (SuperSandro2000) [inline pkgs/applications/backup/vorta/default.nix:36]: Please use substituteInPlace and add the ".

File: pkgs/applications/backup/vorta/default.nix [hunk 2]
```diff
@@ -37,6 +50,38 @@ python3.pkgs.buildPythonApplication rec {
     )
   '';
 
+  checkInputs = with python3Packages; [
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/applications/backup/vorta/default.nix:54]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/backup/vorta/default.nix (hunk 1 above)
Comment (Ma27) [reply pkgs/applications/backup/vorta/default.nix:36]: `substituteInPlace` after `sed` is a no-op here.

Comment (Artturin) [reply pkgs/applications/backup/vorta/default.nix:36]: i forgot to remove the sed

**Pull request merged**